fabric8-ui/fabric8-ui
# fix(auth): auth service for all auth and user API endpoints

Note to the reviewer : 
please use the PR title as the commit message after squashing commits during merge.


- Use `auth.openshift.io` for login+user calls instead of `api.openshift.io`
- update login path from `/api/login/authorize` to `/api/login`


This PR has to be merged with-or-after https://github.com/fabric8-ui/ngx-login-client/pull/86 .

### What this PR does ?
- Introduces the usage of auth service endpoints for existing user/token api calls.
- Reads Auth API url from the environment variable.
- The default is https://prod-preview.openshift.io 
- When tested with https://github.com/fabric8-ui/ngx-login-client/pull/86  , the token for github url would no longer be fetched from keycloak but from the auth service.
- When tested with https://github.com/fabric8-ui/ngx-login-client/pull/86  , the refresh token would be fetched from `https://auth.prod-preview.openshift.io/api/token/refresh` instead of  `https://api.prod-preview.openshift.io/api/login/refresh` .
 ### How to test ?
- Checkout https://github.com/fabric8-ui/ngx-login-client/pull/86 , build it and do an `npm link` from the location where the current fabric8-ui branch has been checked out.
- Do an `npm start`
- Verify that all `/api/user` related urls are constructed with the auth service domain. ( except for /api/user/services ) 
- Verify that github token is not fetched from keycloak.
